js/woo.js

```diff
@@ -0,0 +1,1532 @@
+                    'fullname': this.safeString (item, 'fullname'),
+                    'networks': {},
+                };
+            }
+            const chainedTokenCode = this.safeString (item, 'token');
+            const parts = chainedTokenCode.split ('_');
+            const networkId = parts.length === 2 ? this.safeString (parts, 0) : chainedTokenCode;
+            const networkUnifiedSlug = this.safeString (this.options['network-aliases'], networkId, networkId);
+            derivedCurrenciesData[code]['networks'][networkId] = {
+                'chained_currency_code': chainedTokenCode,
+                'network_exchangeslug': networkUnifiedSlug,
+                'decimals': this.safeString (item, 'decimals'),
+                'info': { 'currencyInfo': item },
+            };
+        }
+        method = this.getSupportedMapping (marketType, {
+            'spot': 'v1PublicGetTokenNetwork',
+        });
+        const response2 = await this[method] (query);
+        const data2 = this.safeValue (response2, 'rows', []);
+        //
+        // {
+        //     rows: [
+        //         {
+        //             protocol: "ERC20",
+        //             token: "USDT",
+        //             name: "Ethereum",
+        //             minimum_withdrawal: 30,
+        //             withdrawal_fee: 25,
+        //             allow_deposit: 1,
+        //             allow_withdraw: 1
+        //         },
+        //         {
+        //             protocol: "TRC20",
+        //             token: "USDT",
+        //             name: "Tron",
+        //             minimum_withdrawal: 30,
+        //             withdrawal_fee: 1,
+        //             allow_deposit: 1,
+        //             allow_withdraw: 1
+        //         },
+        //         ...
+        //     ],
+        //     success: true
+        // }
+        //
+        const derivedNetworksData = {};
+        for (let i = 0; i < data2.length; i++) {
+            const item = data2[i];
+            const tokenCode = this.safeString (item, 'token');
+            const code = this.safeCurrencyCode (tokenCode);
+            if (!(code in derivedNetworksData)) {
+                derivedNetworksData[code] = {
+                    'networks': {},
+                };
+            }
+            const protocol = this.safeString (item, 'protocol', '');
+            const networkUnifiedSlug = this.safeString (this.options['network-aliases'], protocol, protocol);
+            derivedNetworksData[code]['networks'][networkUnifiedSlug] = {
+                'network_title': this.safeString (item, 'name'),
+                'allow_deposit': this.safeString (item, 'allow_deposit'),
+                'allow_withdraw': this.safeString (item, 'allow_withdraw'),
+                'withdrawal_fee': this.safeString (item, 'withdrawal_fee'),
+                'minimum_withdrawal': this.safeString (item, 'minimum_withdrawal'),
+                'info': { 'networkInfo': item },
```

Comment:
I think this code is somewhat an overkill for the purpose. We can do this in an algorithmically-cleaner way.
My logic goes like this:

```JavaScript
        const tokenResponse = await this.v1PublicGetToken (query);
        //  
        //     {
        //         rows: [
        //             {
        //                 token: "ETH_USDT",
        //                 fullname: "Tether",
        //                 decimals: 6,
        //                 balance_token: "USDT",
        //                 created_time: "0",
        //                 updated_time: "0"
        //             },
        //         ],
        //         success: true
        //     }
        //            
        const tokenNetworkResponse = await this.v1PublicGetTokenNetwork (query);
        //
        //     {
        //         rows: [
        //             {
        //                 protocol: "ERC20",
        //                 token: "USDT",
        //                 name: "Ethereum",
        //                 minimum_withdrawal: 30,
        //                 withdrawal_fee: 25,
        //                 allow_deposit: 1,
        //                 allow_withdraw: 1
        //             },
        //             {
        //                 protocol: "TRC20",
        //                 token: "USDT",
        //                 name: "Tron",
        //                 minimum_withdrawal: 30,
        //                 withdrawal_fee: 1,
        //                 allow_deposit: 1,
        //                 allow_withdraw: 1
        //             },
        //             ...
        //         ],
        //         success: true
        //     }
        //
        const result = {};
        const tokenRows = this.safeValue (tokenResponse, 'rows', []);
        const tokenNetworkRows = this.safeValue (tokenNetworkResponse, 'rows', []);
        const networksByCurrencyId = this.groupBy (tokenNetworkRows, 'token');
        for (let i = 0; i < tokenRows.length; i++) {
            //
            //     ** please copy this code from huobi.js as much as possible **
            //     ** i'm not including all the details in this example for simplicity **
            //
            const currency = tokenRows[i];
            const id = this.safeString (currency, 'balance_token');
            const code = this.safeCurrencyCode (id);
            const networks = this.safeValue (networksByCurrencyId, id, []);
            const resultingNetworks = {};
            for (let j = 0; j < networks.length; j++) {
                const networkEntry = networks[i];
                const networkId = this.safeString (networkEntry, 'protocol');
                const network = this.safeNetwork (networkId);
                resultingNetworks[network] = {
                    'info': chain,
                    'id': networkId,
                    'network': network,
                    'limits': {
                        'withdraw': {
                            'min': minWithdraw,
                            'max': maxWithdraw,
                        },
                    },
                    'active': active,
                    'deposit': depositEnabled,
                    'withdraw': withdrawEnabled,
                    'fee': fee,
                    'precision': precision,
                };
            }
            result[code] = {
                // currency structure
                // ...
                'networks': resultingNetworks,
                // ...
            };
        }
```